src/plots/ternary/ternary.js

```diff
@@ -349,6 +347,62 @@ proto.adjustLayout = function(ternaryLayout, graphSize) {
         .style('stroke-width', (caxis.linewidth || 0) + 'px');
 };
 
+proto.drawAxes = function(doTitles) {
+    var _this = this,
+        gd = _this.graphDiv,
+        titlesuffix = _this.id.substr(7) + 'title',
+        aaxis = _this.aaxis,
+        baxis = _this.baxis,
+        caxis = _this.caxis;
+    // 3rd arg true below skips titles, so we can configure them
+    // correctly later on.
+    Axes.doTicks(gd, aaxis, true);
+    Axes.doTicks(gd, baxis, true);
+    Axes.doTicks(gd, caxis, true);
+
+    if(doTitles) {
+        var apad = Math.max(aaxis.showticklabels ? aaxis.tickfont.size / 2 : 0,
+            (caxis.showticklabels ? caxis.tickfont.size * 0.75 : 0) +
+            (caxis.ticks === 'outside' ? caxis.ticklen * 0.87 : 0));
+        Titles.draw(gd, 'a' + titlesuffix, {
+            propContainer: aaxis,
+            propName: _this.id + '.aaxis.title',
+            dfltName: 'Component A',
+            attributes: {
+                x: _this.x0 + _this.w / 2,
+                y: _this.y0 - aaxis.titlefont.size / 3 - apad,
+                'text-anchor': 'middle'
+            }
+        });
+
+        var btitleSize = baxis.titlefont.size,
```

Comment:
Let's either keep the consistency above and define  `var atitleSize = aaxis.titlefont.size`, or use the full object path for `b` and `c`.